zaza/openstack/charm_tests/vault/utils.py

```diff
@@ -76,7 +77,14 @@ def initialize(self):
         else:
             self.vault_creds = init_vault(self.unseal_client)
             store_credentials(self.vault_creds)
-        self.initialized = is_initialized(self.unseal_client)
+
+        for _ in range(10):
+            if is_initialized(self.unseal_client):
+                break
+            time.sleep(10)
```

Comment:
I think using tenacity might be a bit of overkill here.  The method is "initialize" and the outcome of the function should be "it's initialised".  I'm not sure it's worth pulling out a wait-for-initialised function as it would probably only be used from this function.

However, the original code is a bit of a mess; it has a `@property`  `is_initialized` which just returns the attribute `initialized`, which seems like a waste; the intention of using a property would be to do some work to find out if the vault was initialised.  e.g. the property auto to return `is_initialized(self.unseal_client)`; it's all a bit strange.

Reply:
Sure, for some reason, I am not able to mark this comment as resolved (Where's the button gone ?). Please feel free to do so.